src/main/java/picard/analysis/MergeableMetricBase.java

```diff
@@ -143,15 +145,26 @@ public MergeableMetricBase merge(final MergeableMetricBase other) {
         for (final Field field : getAllFields(this.getClass())) {
             if (field.isSynthetic()) continue;
 
-            if (field.getAnnotationsByType(MergeByAdding.class).length +
+            final boolean isAnnotated =
+                    field.getAnnotationsByType(MergeByAdding.class).length +
                     field.getAnnotationsByType(MergeByAssertEquals.class).length +
                     field.getAnnotationsByType(NoMergingIsDerived.class).length +
                     field.getAnnotationsByType(MergingIsManual.class).length +
-                    field.getAnnotationsByType(NoMergingKeepsValue.class).length == 0) {
-                throw new IllegalStateException("All fields of this class must be annotated with @MergeByAdding, @NoMergingIsDerived, or @MergeByAssertEquals. " +
+                    field.getAnnotationsByType(NoMergingKeepsValue.class).length != 0;
+
+            final boolean isStatic = Modifier.isStatic(field.getModifiers());
+            final boolean isAnnotatedActive = field.getAnnotationsByType(MergeByAdding.class).length != 0;
```

Comment:
This code computes both `isAnnotated` and `isAnnotatedActive` but these are only used on a single branch of the `if`. How about:
```
if (Modifier.isStatic(field.getModifiers()) {
    final boolean isAnnotatedMergeByAdding = field.getAnnotationsByType(MergeByAdding.class).length != 0;
    if (isAnnotatedMergeByAdding) {
        throw new IllegalStateException(...);
    }
} else {
    final boolean isAnnotated = ...;
    if (!isAnnotated) {
        throw new IllegalStateException(...);
    }
}
```
This also flips the order of the ifs and renames `isAnnotatedActive` since that name doesn't seem to reflect its value.